routes/api/v1.js

```diff
@@ -1079,6 +1079,88 @@ api.declare({
 });
 
 
+/** Report task canceled */
+api.declare({
+  method:     'post',
+  route:      '/task/:taskId/runs/:runId/canceled',
+  name:       'reportCanceled',
```

Comment:
Hmm... I'm not sure we should require a `runId` to be specified. After all don't we just want to cancel the latest run, regardless of it's `runId`.... Calling `queue.status` to get the `runId` and then calling `queue.reportCanceled` opens the possibility that a new run is created in between the two calls.. Note, the semantics are different based on whether or not we specify `runId`, so the argument for can definitely be made for both.

But I think the semantics we're looking for here is that a scheduler (task-graph scheduler) or person who created a task want's to cancel it, regardless of the current `runId`. Hence, I suggest route and name as follows:
```
  route:  '/task/:taskid/cancel',
  name: 'cancelTask'
```